com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/LaunchRequestHandler.java

```diff
@@ -166,7 +176,7 @@
             ListeningConnector listenConnector = connectors.get(0);
```

Comment:
listenConnector is not needed for NO_DEBUG mode. Actually I think it would be better to extract the common part instead of using lots of ifs